jobs/bump-jenkins-plugins.Jenkinsfile

```diff
@@ -0,0 +1,127 @@
+    checkout scm
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+    pipecfg = pipeutils.load_pipecfg()
+}
+
+properties([
+    pipelineTriggers([
+        // check once a month
+        pollSCM('H H 1 * *')
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+repo = "coreosbot-releng/fedora-coreos-pipeline"
+botCreds = "github-coreosbot-releng-token-username-password"
+pr_branch = "pluginsupdate"
+
+def getPluginLatestVersion(pluginsUpdate) {
+    // Extract the plugin version from the URL 
+    def versionPattern = /\/([^\/]+)\/([^\/]+)\/([^\/]+)\.hpi/
+    def matcher = (pluginsUpdate =~ versionPattern)
+    def pluginVersion
+
+    if (matcher.find()) {
+        def groupId = matcher.group(1)
+        pluginVersion = matcher.group(2)
+    } else {
+        println "Unable to extract plugin version from the URL."
+    }
+    return pluginVersion
+}
+
+lock(resource: "bump-jenkins") {
+    node{
+        try {
+            shwrap("""
+                git config --global user.name "CoreOS Bot Releng"
+                git config --global user.email "coreosbot-releng@fedoraproject.org"
+            """)
+
+            def pluginslist
+            def pluginsToUpdate = [:]
+            def haveChanges=false 
+
+            stage("Read plugins.txt") {
+                shwrapCapture("""
+                    if [[ -d fedora-coreos-pipeline ]]; then
+                        rm -rf fedora-coreos-pipeline
+                    fi
+                    git clone --branch pluginsupdate https://github.com/coreosbot-releng/fedora-coreos-pipeline.git
+                    cd fedora-coreos-pipeline
+
+                    # Check if the branch exists
+                    if git ls-remote --heads origin ${pr_branch} | grep ${pr_branch}; then
+                        git checkout ${pr_branch}
+                    else
+                        git checkout -b ${pr_branch}
+                    fi
+                """)
+                def plugins_lockfile = "jenkins/controller/plugins.txt"
+                pluginslist = shwrapCapture("cat $plugins_lockfile | grep -v ^#").split('\n')
+            }
+
+            stage("Check for plugin updates") {
+                def pluginsUpdate
+                pluginslist.each { plugin ->
+                    def parts = plugin.split(':')
+                    if (parts.size() == 2) {
+                        def pluginName = parts[0]
+                        def currentVersion = parts[1]
+                        pluginsUpdate = shwrapCapture("curl -Ls -o /dev/null -w '%{url_effective}' https://updates.jenkins.io/download/plugins/${pluginName}/latest/${pluginName}.hpi")
+                        def latestVersion = getPluginLatestVersion(pluginsUpdate)
+                        if (latestVersion.toString() != currentVersion.toString()) {
+                            haveChanges = true
```

Comment:
Instead of tracking whether we have changes in a variable, we can use `git diff --exit-code` to know if we did changes.

Reply:
Removed the variable and used `git diff --exit-code` instead